jdaviz/configs/specviz/plugins/parsers.py

```diff
@@ -50,6 +50,11 @@ def specviz_spectrum1d_parser(app, data, data_label=None, format=None, show_in_v
             except IORegistryError:
                 # Multi-extension files may throw a registry error
                 data = SpectrumList.read(str(path), format=format)
+        elif path.is_dir():
+            data = SpectrumList.read(str(path), format=format)
+            # temporary labelling, just for testing. Better labels should
+            # come from the file list
+            data_label = ["Spectrum " + str(i) for i in range(len(data))]
```

Comment:
Nitpick... though a deeper issue is this line ignores any user-provided `data_label`... 🤷 
```suggestion
            data_label = [f"Spectrum {i}" for i in range(len(data))]
```

Reply:
Indeed, you're right. The code is missing a way to allow the user to define the labels. I will fix that...